spotify_player/src/client/mod.rs

```diff
@@ -1305,7 +1305,13 @@ impl Client {
             let new_track = prev_track_name != curr_track_name && !curr_track_name.is_empty();
             // check if we need to update the buffered playback
             let needs_update = match (&player.buffered_playback, &player.playback) {
-                (Some(bp), Some(p)) => bp.device_id != p.device.id || new_track,
+                (Some(bp), Some(p)) => {
+                    bp.device_id != p.device.id
+                        || bp.volume != p.device.volume_percent
+                        || bp.shuffle_state != p.shuffle_state
+                        || bp.repeat_state != p.repeat_state
+                        || new_track
```

Comment:
this is not ideal because Spotify takes time to update the playback. The first `shuffle_state`  reported by Spotify API after changing shuffle state will be the old `shuffle_state`. As a result, the **modified** buffered playback will be reset to the old state.

I tried another approach in https://github.com/aome510/spotify-player/pull/341. Can you check if that fixes your issue?

Reply:
Thank you for working on this. While the approach in #341 does fix the issue when the `playback_refresh_duration_in_ms` > 0, it introduces a flickering of the current playback as can be seen [here](https://asciinema.org/a/tX3vrbY7Kekd8F6HPaM11Mf62).

Reply:
update the branch with additional fix for the flickering. You'll need to **force reset** the branch with
```
git fetch && git reset --hard origin
```

Reply:
This works perfectly, thank you so much for taking the time!

I'll close this PR so you can accept your own.